arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -312,6 +302,75 @@ private void abortingCall(Hub hub) {
     commonValues.collectChildStipend(hub);
   }
 
+  /**
+   * Sets the scenario to the relevant undefined variant, i.e. either
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_PRC_UNDEFINED}
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_SMC_UNDEFINED}
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_EOA_UNDEFINED}
+   *
+   * <p>depending on the address.
+   *
+   * @param hub
+   */
+  private void refineUndefinedScenario(Hub hub) {
+
+    final boolean aborts = hub.pch().abortingConditions().any();
+    if (aborts) {
+      scenarioFragment.setScenario(CALL_ABORT_WONT_REVERT);
+      return;
+    }
+
+    final WorldUpdater world = hub.currentFrame().frame().getWorldUpdater();
+    if (isPrecompile(calleeAddress)) {
+      precompileAddress = Optional.of(calleeAddress);
+      scenarioFragment.setScenario(CALL_PRC_UNDEFINED);
+      // Account rows for precompile are traced at contextReEntry
+
+      precompileSubsection =
+          ADDRESS_TO_PRECOMPILE.get(preOpcodeCalleeSnapshot.address()).apply(hub, this);
+    } else {
+      Optional.ofNullable(world.get(calleeAddress))
+          .ifPresentOrElse(
+              account -> {
+                scenarioFragment.setScenario(
+                    account.hasCode() ? CALL_SMC_UNDEFINED : CALL_EOA_UNDEFINED);
+              },
+              () -> {
+                scenarioFragment.setScenario(CALL_EOA_UNDEFINED);
+              });
+    }
+  }
+
+  private void eoaProcessing(Hub hub) {
+    hub.defers().scheduleForContextReEntry(this, hub.currentFrame());
+    commonValues.collectChildStipend(hub);
+    finalContextFragment = ContextFragment.nonExecutionProvidesEmptyReturnData(hub);
+  }
+
+  private void smcProcessing(Hub hub, MessageFrame frame) {
+    CallFrame currentFrame = hub.currentFrame();
+    hub.defers().scheduleForContextEntry(this);
+    hub.defers().scheduleForContextExit(this, hub.callStack().futureId());
+    hub.defers().scheduleForContextReEntry(this, currentFrame);
+
+    hub.defers().scheduleForContextReEntry(firstImcFragment, currentFrame);
+
+    this.commonValues.payGasPaidOutOfPocket(hub);
+    finalContextFragment = ContextFragment.initializeNewExecutionContext(hub);
+    hub.romLex().callRomLex(frame);
+  }
+
+  private void prcProcessing(Hub hub) {
+    hub.defers().scheduleForContextEntry(this);
```

Comment:
do the stuff here, instead of in the `setScenario()' method

Reply:
I'm not understanding.

Reply:
talking about this :

precompileAddress = Optional.of(calleeAddress);
precompileSubsection =  ADDRESS_TO_PRECOMPILE.get(preOpcodeCalleeSnapshot.address()).apply(hub, this);